kms/spec/kms_spec.rb

```diff
@@ -0,0 +1,367 @@
+          member:, role:, location:)
+    kms_client = create_service_client
+
+    resource = "projects/#{project_id}/locations/#{location}/" +
+               "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+    policy = kms_client.get_project_location_key_ring_crypto_key_iam_policy resource
+
+    policy.bindings ||= []
+    policy.bindings << Google::Apis::CloudkmsV1beta1::Binding.new(members: [member],
+                                                                  role: role)
+    kms_client.set_crypto_key_iam_policy resource, policy_request
+  end
+
+  def test_add_member_to_keyring_policy(project_id:, key_ring_id:, member:,
+        role:, location:)
+    kms_client = create_service_client
+
+    policy = test_get_keyring_policy(project_id: project_id, key_ring_id: key_ring_id,
+         location: location)
+
+    resource = "projects/#{project_id}/locations/#{location}/" +
+        "keyRings/#{key_ring_id}"
+
+
+    policy.bindings ||= []
+    policy.bindings << Google::Apis::CloudkmsV1beta1::Binding.new(members: [member],
+                                                                  role: role)
+
+    policy_request = Google::Apis::CloudkmsV1beta1::SetIamPolicyRequest.new(policy: policy)
+
+    kms_client.set_key_ring_iam_policy resource, policy_request
+  end
+
+  def test_encrypt_file(project_id:, key_ring_id:, crypto_key:, location:,
+      input_file:, output_file:)
+    kms_client = create_service_client
+
+    name = "projects/#{project_id}/locations/#{location}/" +
+      "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+    encoded_file = File.read(input_file).chomp
+
+    request = Google::Apis::CloudkmsV1beta1::EncryptRequest.new
+    request.plaintext = encoded_file
+
+    response = kms_client.encrypt_crypto_key name, request
+
+    File.write(output_file, response.ciphertext)
+  end
+
+  def test_decrypt_file(project_id:, key_ring_id:, crypto_key:, location:,
+      input_file:, output_file:)
+    kms_client = create_service_client
+
+    name = "projects/#{project_id}/locations/#{location}/" +
+      "keyRings/#{key_ring_id}/cryptoKeys/#{crypto_key}"
+
+    encoded_file = File.read(input_file).chomp
+
+    request = Google::Apis::CloudkmsV1beta1::DecryptRequest.new
+    request.ciphertext = encoded_file
+
+    response = kms_client.decrypt_crypto_key name, request
+
+    File.write(output_file, response.plaintext)
+  end
+
+  before :all do
+    @project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+    @key_ring_id = "#{@project_id}_key_ring_#{Time.now.to_i}"
+    @cryptokey_id = "#{@project_id}_cryptokey_#{Time.now.to_i}"
+    @location = "global"
+
+    @test_key_ring = test_create_keyring project_id: @project_id,
+        key_ring_id: @key_ring_id, location: @location
+
+    @test_cryptokey = test_create_cryptokey project_id: @project_id,
+        key_ring_id: @key_ring_id, crypto_key: @cryptokey_id, location: @location
+
+    @input_file = File.expand_path "resources/file.txt", __dir__
+
+    $VERBOSE = nil
```

Comment:
Let's add a comment to note why we're doing this.  Because seeing `$VERBOSE = nil` is kindof a scary thing to see in someone's Ruby code 😛